domain/ide/command/ignores_request.go

```diff
@@ -0,0 +1,289 @@
+	if !ok {
+		return nil, fmt.Errorf("reason should be a string")
+	}
+	expiration, ok := cmd.command.Arguments[4].(string)
+	if !ok {
+		return nil, fmt.Errorf("expiration should be a string")
+	}
+	ignoreId, ok := cmd.command.Arguments[5].(string)
+	if !ok {
+		return nil, fmt.Errorf("ignoreId should be a string")
+	}
+
+	gafConfig := engine.GetConfiguration().Clone()
+	gafConfig.Set(ignore_workflow.FindingsIdKey, findingsId)
+	gafConfig.Set(ignore_workflow.IgnoreTypeKey, ignoreType)
+	gafConfig.Set(ignore_workflow.ReasonKey, reason)
+	gafConfig.Set(ignore_workflow.ExpirationKey, expiration)
+	gafConfig.Set(ignore_workflow.EnrichResponseKey, true)
+	gafConfig.Set(ignore_workflow.InteractiveKey, false)
+	gafConfig.Set(ignore_workflow.IgnoreIdKey, ignoreId)
+	gafConfig.Set(configuration.INPUT_DIRECTORY, contentRoot)
+	return gafConfig, nil
+}
+
+func (cmd *submitIgnoreRequest) deleteIgnoreRequest(engine workflow.Engine, findingsId string, contentRoot types.FilePath, issue types.Issue) error {
+	gafConfig, err := cmd.createDeleteConfiguration(engine, findingsId, contentRoot)
+	if err != nil {
+		return err
+	}
+
+	response, err := engine.InvokeWithConfig(ignore_workflow.WORKFLOWID_IGNORE_CREATE, gafConfig)
+	if err != nil {
+		return err
+	}
+
+	if len(response) == 0 {
+		return fmt.Errorf("no data returned from ignore workflow")
+	}
+
+	output, ok := response[0].GetPayload().([]byte)
+	if !ok {
+		return fmt.Errorf("invalid response from ignore workflow")
+	}
+
+	err = updateIssueWithIgnoreDetails(cmd.c, output, issue)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cmd *submitIgnoreRequest) createDeleteConfiguration(engine workflow.Engine, findingsId string, contentRoot types.FilePath) (configuration.Configuration, error) {
+	if len(cmd.command.Arguments) < 3 {
+		return nil, fmt.Errorf("insufficient arguments for ignore-delete workflow")
+	}
+
+	ignoreId, ok := cmd.command.Arguments[5].(string)
+	if !ok {
+		return nil, fmt.Errorf("ignoreId should be a string")
+	}
+
+	gafConfig := engine.GetConfiguration().Clone()
+	gafConfig.Set(ignore_workflow.FindingsIdKey, findingsId) //TODO remove this one?
+	gafConfig.Set(ignore_workflow.IgnoreIdKey, ignoreId)
+	gafConfig.Set(ignore_workflow.EnrichResponseKey, true)
+	gafConfig.Set(ignore_workflow.InteractiveKey, false)
+	gafConfig.Set(configuration.INPUT_DIRECTORY, contentRoot)
+	return gafConfig, nil
+}
+
+func updateIssueWithIgnoreDetails(c *config.Config, output []byte, issue types.Issue) error {
+	var suppression sarif.Suppression
+	err := json.Unmarshal(output, &suppression)
+	if err != nil {
+		return err
+	}
+	isIgnored, ignoreDetails := code.GetIgnoreDetailsFromSuppressions(c, []sarif.Suppression{suppression})
+
+	issue.SetIsIgnored(isIgnored)
+	issue.SetIgnoreDetails(ignoreDetails)
+	return nil
+}
+
+func (cmd *submitIgnoreRequest) sendShowDocumentRequest(issue types.Issue) {
+	snykUri, _ := code.SnykMagnetUri(issue, code.ShowInDetailPanelIdeCommand)
```

Comment:
please refactor, this seems like code duplication with autofix. also, errors must be handled.